### ISSUE TITLE
please export colors as variables

### ISSUE THREAD

**Issue by wbolster** [bbatsov/solarized-emacs#277]
exporting colors as variables helps when reusing the colours elsewhere, e.g. for customization in `init.el`. something like this:

``` elisp
(setq
 solarized-color-yellow  "#b58900"
 solarized-color-orange  "#cb4b16"
 solarized-color-red     "#dc322f"
 solarized-color-magenta "#d33682"
 solarized-color-violet  "#6c71c4"
 solarized-color-blue    "#268bd2"
 solarized-color-cyan    "#2aa198"
 solarized-color-green   "#859900")
```

...but with `(defvar ...)`.

**Comment by thomasf** [comment]
It could be done inside the themes `custom-theme-set-variables` phase but that would only make the colors usable after the theme is loaded but that might be a good thing since the baseXX colors can be inverted correctly depending on bright/dark mode.

**Comment by dcherian** [comment]
Yeah that would work great

**Comment by wbolster** [comment]
awesome. though i toggle dark/light, for me the most important part is the actual colours. i use them to tweak cursor appearance based on the evil state (insert, normal, visual, and so on).

**Comment by jgkamat** [comment]
If possible, I would love a way to customize the actual colors being used. For example, all  the colors listed [here](https://github.com/bbatsov/solarized-emacs/blob/master/solarized.el#L140-L181) are hard-coded in the source, if they were put into a table we could customize before solarized is loaded, we could build solarized themes with tweaked colors (not tweaking a single face, as the child system does right now). 

Right now I'm using a very old forked version from a long time ago with my colors, but I'd much rather follow and contribute to upstream. I think if this is done right, it could make the colors accessible in general and allow us to tweak the base colors of the theme.

**Comment by conao3** [comment]
Please try `solarized-color-palette-alist` variable and `solarized-create-theme-with-palette` feature merged at #330
Enjoy!

**Comment by wbolster** [comment]
thanks for the heads up, @conao3. two years of silence, and suddenly there is progress. 🚀

 thanks for working on this package!
